lib/shared/sdk/usbboot.js

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2017 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This work is heavily based on https://github.com/raspberrypi/usbboot
+ * Copyright 2016 Raspberry Pi Foundation
+ */
+
+'use strict'
+
+const _ = require('lodash')
+const Bluebird = require('bluebird')
+const usb = require('usb')
+
+/**
+ * @summary The number of USB interfaces in devices with a BCM2835 SoC
```

Comment:
I wonder if this could be clarified somehow to indicate that this is the number of interfaces in the USB endpoint, and not the physical number of USB ports that the SoC actually has? Doesn't matter if not, only a very minor nit.